### DISCUSSION TITLE
ADFS with NPM

### DISCUSSION THREAD

**Discussion by jeremy-chua** [NginxProxyManager/nginx-proxy-manager#2628] [Q&A]
I have ADFS setup with a self signed certificate and behind NPM with Let's Encrypt certs.
That access to the SSO page fails using IP address as Ngnix uses resolver instead of /etc/hosts file. Hence i set the DNS in docker-compose file with "dns" option. Nignix is able to reach the adfs local FDQN.

I'm having an issue when loading the SSO page with HTTP status 400. I googled and most of the nginx user solution is to include the following settings:

`
	proxy_set_header	X-Real-IP $remote_addr;
	proxy_set_header	X-Forwarded-For $proxy_add_x_forwarded_for;
	proxy_set_header	Host $http_host;
	proxy_set_header 	X-Forwarded-Proto $scheme;
	proxy_redirect		off;

`
Reference: 
https://peter.upfold.org.uk/blog/2016/05/31/reverse-proxying-adfs-with-nginx/
https://forum.nginx.org/read.php?11,274340,281719
https://blog.jamiebaldanza.org/2016/03/22/nginx-reverse-proxy-adfs-3-0-server/

BUT i'm still getting a 400, just wondering if any NPMer managed to have ADFS (windows server 2022) worked with NPM?

**Comment by Sopraner** [comment]
It took 3 days to find the answer! 😅

Tested on Ubuntu 22 and Synology 7 with the last version of Nginx Proxy Manager in the docker.
Default AD FS setup (Windows Server 2022) with SSO. No DNS changes.
NPM settings
"Websocket Support" set to OFF. SSL - force ON. Advanced tab:

ssl_protocols TLSv1.2 TLSv1.3;
proxy_ssl_name $host;
proxy_ssl_server_name on;
fastcgi_buffers 16 16k;
fastcgi_buffer_size 32k;
proxy_buffer_size   128k;
proxy_buffers   4 256k;
proxy_busy_buffers_size   256k;

And that's all!
Also thanks to these guys.
https://stackoverflow.com/questions/42613491/azure-ad-webapp-behind-reverse-proxy-receives-502-bad-gateway

**Comment by flaktivated** [comment]
Thanks for the guidance here - I have tried both cofnigurations and am still getting the 400 Response code.
I am trying to move from HAProxy (whcch works fine with no special configuration outside of forwarding host).
